seirmo/plots/_plot_from_numpy.py

```diff
@@ -0,0 +1,189 @@
+        self._size = subplots_columns * subplots_rows
+        self._nrows = subplots_rows
+        self._ncolumns = subplots_columns
+        # we store a figure object and multiple axes objects
+
+        # Ensure self._axes array is always 2D
+        if self._nrows == 1 and self._ncolumns == 1:
+            self._axes = np.array(self._axes)[np.newaxis, np.newaxis]
+        elif self._nrows == 1:
+            self._axes = np.array(self._axes)[np.newaxis, :]
+        elif self._ncolumns == 1:
+            self._axes = np.array(self._axes)[:, np.newaxis]
+
+    def __getitem__(self, index):
+        """If figure = ConfigurablePlotter(), then figure.begin().
+        Figure[0] will return the matplot figure, and figure[1] will
+        return the subplot axis objects"""
+        if index == 0:
+            return self._fig
+        elif index == 1:
+            return self._axes
+        else:
+            raise ValueError("Index must be 0 (for figure) or 1 (for axes)")
+
+    def add_data_to_plot(
+        self,
+        times: np.ndarray,
+        data_array: np.ndarray,
+        position: list = [0, 0],
+        xlabel: str = "time",
+        ylabels: list = [],
+        colours: list = [],
+        new_axis=False,
+    ):
+        """Main code to add new data into the plot
+
+        :params:: times: np.ndarray, independant x- variable
+        :params:: data_array: np.ndarray, multiple dependent y- variables
+                              Data should has one row per timestep,
+                              and one column for each dependent variable
+        :params:: position: list of integers, gives index of subplot to use
+        :params:: xlabel: str
+        :params:: ylabel: list of strings (a single string is also accepted)
+        :params:: colours: list of valid colour specifiers (ie strings or
+                           rgb tuples)
+        :params:: new_axis: boolean, set to true if data should
+                            be plotted on a second x axis"""
+
+        if len(data_array.shape) == 1:  # Turn any 1D input into 2D
+            if type(times) != np.ndarray or np.sum(np.shape(times)) == 1:
+                # I.e. if only one np.int, or one element array
+                times = np.array(times, ndmin=2)
+                data_array = data_array[np.newaxis, :]
+            else:
+                data_array = data_array[:, np.newaxis]
+
+        assert (
+            times.shape[0] == data_array.shape[0]
+        ), "data and times are not the same length"
+        data_width = data_array.shape[1]  # saves the number of y-var
+
+        assert (
+            position[0] < self._nrows and position[1] < self._ncolumns
+        ), "position and shape are not compatible"
+
+        if new_axis:
+            axis = self._axes[position[0], position[1]].twinx()
+        else:
+            axis = self._axes[position[0], position[1]]
+
+        # Format user inputs
+        if not isinstance(colours, list):
```

Comment:
```suggestion
        if isinstance(colours, str):
```
I see what you are doing, but what is the reason that you want to force a list for this one?

The usual python convention is to not care whether they send ['red', 'blue'] or ('red', 'blue') or np.array(['red', 'blue']) or anything similar. As long as it's not a string (which we need to deal with separately here), it has the correct length, and we can do colours[i], we should tolerate it unless there's a special reason not to.